### PR TITLE
Build and push SNAPSHOT operator images

### PR DESCRIPTION
**Pull request by robobario**
### Type of change

- Enhancement / new feature

### Description

We have a PR beginning to add system tests for the operator but we want to pull an operator image from quay. This PR starts publishing snapshot operator images to quay.

Note that we need a quay repo and to update some variables in github actions.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Update documentation
- [ ] Make sure all unit/integration tests pass
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] If applicable to the change, [trigger the system test suite](../blob/main/DEV_GUIDE.md#jenkins-pipeline-for-system-tests).  Make sure tests pass.
- [ ] If applicable to the change, [trigger the performance test suite](../blob/main/PERFORMANCE.md#jenkins-pipeline-for-performance). Ensure that any degradations to performance numbers are understood and justified.
- [ ] Ensure the PR references relevant issue(s) so they are closed on merging.
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

> **_NOTE:_**  You must be a member of `@kroxylicious/developers` to trigger the system test and performance test suites.  If you are not part of this group, comment on the PR requesting a trigger, tagging `@kroxylicious/developers`.

### CODE REVIEW

**Comment by robobario** [comment]
tested on my fork with a snapshot push https://github.com/robobario/kroxylicious/actions/runs/12385213726

will try pushing up a tag

**Comment by robobario** [comment]
currently the operator manifests refer to `quay.io/kroxylicious/operator:latest`, wondering if we should publish `x.y.z-SNAPSHOT` and `latest` or pick one way.

**Comment by SamBarker** [comment]
> currently the operator manifests refer to `quay.io/kroxylicious/operator:latest`, wondering if we should publish `x.y.z-SNAPSHOT` and `latest` or pick one way.

I generally think using latest is a poor call but I think that ship has sailed and it’s expected. Let’s do both

**Comment by robobario** [comment]
alrighty, it now emits both tags

**Comment by robobario** [comment]
and here's an example [tag push build](https://github.com/robobario/kroxylicious/actions/runs/12385402490/job/34571576578) showing it currently doesn't push anything on tag push

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -35,21 +35,35 @@ jobs:
...
           context: .
           platforms: linux/amd64,linux/arm64
           push: true
           build-args: |
             KROXYLICIOUS_VERSION=${{ env.RELEASE_VERSION }} 
-          tags: ${{ vars.REGISTRY_DESTINATION }}:${{ env.RELEASE_VERSION }}
+          tags: ${{ env.PROXY_IMAGE }}:${{ env.RELEASE_VERSION }}
+      - name: Maven build operator code
+        run: |
+          mvn package --projects :kroxylicious-operator --also-make -DskipTests -B
```
Comment (k-wall) [inline .github/workflows/docker.yaml:60]:
```suggestion
          mvn package --projects :kroxylicious-operator --also-make -Dquick -B 
```

**Comment by robobario** [comment]
I've have pushed up moving the build into the Dockerfile to align it with the proxy image (move the maven build into the image). To do this I've lifted the Dockerfile up into the top dir since it needs to have the whole maven project in it's context. I think that if we left it in the `kroxylicious-operator` directory it would surprise people that `docker build .` fails.

**Review by k-wall — APPROVED** [review]
Not tested, but lgtm.  Thanks @robobario

**Comment by robobario** [comment]
currently failing https://github.com/robobario/kroxylicious/actions/runs/12402177592 , looks like building with `--projects xyz --also-make` is causing trouble, I don't think it's aware it needs to build to bom. I'll have it do a full build of the project for now.

**Pull request merged**